sklearn/linear_model/stochastic_gradient.py

```diff
@@ -355,15 +345,15 @@ def _partial_fit(self, X, y, alpha, C,
                                                            self.classes_, y)
         sample_weight = self._validate_sample_weight(sample_weight, n_samples)
 
-        if self.coef_ is None or coef_init is not None:
+        if getattr(self, "coef_", None) is None or coef_init is not None:
```

Comment:
Can it still happen that ``coef_`` is ``None``? Or is it always "not present"? Then you could use ``hasattr``... I find attributes with trailing underscores that are ``None`` somewhat odd.